server/sonar-process/src/main/java/org/sonar/process/NetworkUtils.java

```diff
@@ -37,6 +47,44 @@ public static int freePort() {
     return RANDOM_PORT_FINDER.getNextAvailablePort();
   }
 
+  /**
+   * Identifying the localhost machine
+   * It will try to retrieve the hostname and the IPv4 addresses
+   *
+   * @return "hostname (ipv4_1, ipv4_2...)"
+   */
+  public static String getHostName() {
+    String hostname;
+    String ips;
+    try {
+      hostname = InetAddress.getLocalHost().getHostName();
+    } catch (UnknownHostException e) {
+      hostname = "unresolved hostname";
+    }
+
+    try {
+      ips = list(NetworkInterface.getNetworkInterfaces()).stream()
+        .flatMap(netif ->
+          list(netif.getInetAddresses()).stream()
+            .filter(inetAddress ->
+              // Removing IPv6 for the time being
+              inetAddress instanceof Inet4Address &&
+                // Removing loopback addresses, useless for identifying a server
+                !inetAddress.isLoopbackAddress() &&
+                // Removing interfaces without IPs
+                !isBlank(inetAddress.getHostAddress())
+            )
+            .map(InetAddress::getHostAddress)
+        )
+        .filter(p -> !isBlank(p))
+        .collect(Collectors.joining(","));
+    } catch (SocketException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)